src_blog/_posts/Grain-Formatter.md

```diff
@@ -0,0 +1,120 @@
+
+This really helps codebases and teams in several ways. It reduces arguments at code review time, as the correct style is guaranteed. It helps new members of the team learn the coding style, and it keeps a very consistent code base that’s easy to read by everyone. My personal favourite is I can be really lazy as I type code, and with `Format on Save` enabled, I just press `Save` and watch the code become well structured.
+
+
+## Grainformat
+
+The formatter can be invoked using the Grain CLI
+
+```sh
+grain format myfile.gr
+```
+
+or 
+
+```sh
+grain format  
+``` 
+
+which will read from stdin, which is generally useful for tools like the [Grain Language Server (LSP)](https://github.com/grain-lang/grain-language-server).
+
+Both write the reformatted code to stdout.
+
+Coming soon will be a check mode and an in-place rewrite mode.  Check mode will scan code and determine if it matches the formatter style without reformatting — useful for CI checks to ensure all code is properly formatted.
+
+The next version of the LSP in our VSCode language extension for Grain to be released alongside the formatter supports whole document formatting, so if you enable `Format On Save` it will reformat all the code each time you save. Code section formatting will be in a minor release after that.
+
+
+## Example of formatting choices
+
+The primary role of the formatter is to make sure no line exceeds the specified maximum length and to break code across multiple lines in a consistent way. Much of getting the formatter right is choosing where those breaks should come.
+
+We've had some help here—we use the fantastic pretty printer that ships with the [ReScript syntax](https://github.com/rescript-lang/syntax).  The primary job of the Grain formatter is to map the Grain compiler output into the intermediate language that the pretty printer understands, where it does all the hard work of calculating what can fit onto one line and breaking it as needed.
+
+The main formatting rules have to do with code indentation. This is what really helps make the code base look consistent. Always 2 spaces (and let’s not get started on tabs vs spaces)!
+
+Curly brace usage in block layouts is another important stylistic choice. The most common styles are Egyptian versus C-style, and we’ve gone for Egyptian:
+
+```sh
+if (conditional) {
+ ...
+} else {
+ ...
+}
+```
+
+There are many choices when it comes to spacing around lists of items, be that function parameters, tuples, lists, arrays, record entries, enums, etc. We add spaces after the separating comments, and before items in some cases. When a list of items breaks over multiple lines, we add a trailing comma to make adding extra items easier:
+
+```sh
+(a,b) and [1,2,3] and  {name:"Super",    value:"Grain"}
```

Comment:
Grain uses `&&` 😛 

Confirmed that this does not work with the formatter!
```suggestion
(a,b) && [1,2,3] &&  {name:"Super",    value:"Grain"}
```

Reply:
This was pseudocode, I was too lazy to make multiple lines but I can fix

Reply:
as you have :-D